src/platform/OpenThread/GenericNetworkCommissioningThreadDriver.cpp

```diff
@@ -27,13 +29,20 @@
 
 using namespace chip;
 using namespace chip::Thread;
+using namespace chip::DeviceLayer::PersistedStorage;
 
 namespace chip {
 namespace DeviceLayer {
 namespace NetworkCommissioning {
-// NOTE: For ThreadDriver, we use two network configs, one is mSavedNetwork, and another is mStagingNetwork, during init, it will
-// load the network config from thread persistent info, and loads it into both mSavedNetwork and mStagingNetwork. When updating the
-// networks, all changes are made on the staging network. When validated we can commit it and save it to the persistent info
+
+// NOTE: For GenericThreadDriver, we assume that the network configuration is persisted by
+// the OpenThread stack after ConnectNetwork command is called, and before that, all the changes
+// are made to a local copy of the dataset stored in mStagingNetwork.
+// Also, in order to support the fail-safe mechanism, the configuration is backed up in a temporary
+// KVS slot upon any changes and restored when the fail-safe timeout is triggered or the device
+// reboots without completing all the changes.
+
+constexpr uint8_t kEmptyDataset[1] = {};
```

Comment:
Should the empty data set comment for the sake of KVS storage be placed here instead?
Wondering if we should have an explanation of what an empty dataset is - it seems we consider a 1-byte that is 0 to be a 'empty dataset' and we rely that this is invalid. Is this correct?

Reply:
Yes, I'll add the comment. Basically, we need to store empty dataset in the backup to be able to revert to "no Thread configuration" on the failsafe expiry. I found that some KVS implementations don't accept empty values and I found it easier to workaround it here than fix all the platforms (also, not sure if there's a simple way to fix that).